src/main/java/fi/dy/masa/servux/dataproviders/StructureDataProvider.java

```diff
@@ -384,10 +463,21 @@ protected void sendStructures(ServerPlayerEntity player,
             NbtList structureList = this.getStructureList(starts, world);
             // System.out.printf("sendStructures: starts: %d -> structureList: %d. refs: %s\n", starts.size(), structureList.size(), references.keySet());
 
-            NbtCompound tag = new NbtCompound();
-            tag.put("Structures", structureList);
+            if (this.registeredPlayers.containsKey(player.getUuid()))
+            {
+                NbtCompound tag = new NbtCompound();
+                tag.put("Structures", structureList);
+                tag.putInt("packetType", ServuxStructuresHandler.PACKET_S2C_STRUCTURE_DATA);
 
-            PacketSplitter.sendPacketTypeAndCompound(StructureDataPacketHandler.CHANNEL, StructureDataPacketHandler.PACKET_S2C_STRUCTURE_DATA, tag, player);
+                if (useApi)
+                {
+                    HANDLER.encodeNbtCompound(player, tag);
+                }
+                else
+                {
+                    HANDLER.sendPlayPayload(player.networkHandler, new ServuxStructuresPayload(tag));
+                }
+            }
```

Comment:
Perhaps, the `useApi` could be renamed to better resemble the fact that it's counting failure attempts and then cancels furher send attempts?

Maybe something along these `ignoreFailure` or `useCountingEncoder`

This could also be applied to the [`ServuxStructuresHandler#encodeNbtCompound()`](https://github.com/sakura-ryoko/servux/blob/b2a63a58292bcfab3548af30e0c465defb1a4ec8/src/main/java/fi/dy/masa/servux/network/handler/ServuxStructuresHandler.java#L113) method

Reply:
the useApi flag has nothing to do with counting failures, it's a flag to change which method it uses to send the data.